test/integration/apiserver/flowcontrol/concurrency_util_test.go

```diff
@@ -0,0 +1,496 @@
+	var noxu1ClientRequestLatencySum float64
+	var noxu1ClientRequestLatencySumSq float64
+	var noxu1ClientRequestLatencyCount int32
+	var noxu1Mutex sync.Mutex
+	streamRequestsWithIndex(noxu1NumGoroutines, func(idx int) {
+		start := time.Now()
+		_, err := noxu1Client.CoreV1().Namespaces().Get(context.Background(), "default", metav1.GetOptions{})
+		duration := time.Since(start).Seconds()
+		noxu1Mutex.Lock()
+		noxu1ClientRequestLatencyCount += 1
+		noxu1ClientRequestLatencySum += duration
+		noxu1ClientRequestLatencySumSq += duration * duration
+		noxu1Mutex.Unlock()
+		if err != nil {
+			t.Error(err)
+		}
+	}, &wg, stopCh)
+	// "mouse"
+	noxu2NumGoroutines := 3
+	var noxu2ClientRequestLatencySum float64
+	var noxu2ClientRequestLatencySumSq float64
+	var noxu2ClientRequestLatencyCount int32
+	var noxu2Mutex sync.Mutex
+	streamRequestsWithIndex(noxu2NumGoroutines, func(idx int) {
+		start := time.Now()
+		_, err := noxu2Client.CoreV1().Namespaces().Get(context.Background(), "default", metav1.GetOptions{})
+		duration := time.Since(start).Seconds()
+		noxu2Mutex.Lock()
+		noxu2ClientRequestLatencyCount += 1
+		noxu2ClientRequestLatencySum += duration
+		noxu2ClientRequestLatencySumSq += duration * duration
+		noxu2Mutex.Unlock()
+		if err != nil {
+			t.Error(err)
+		}
+	}, &wg, stopCh)
+
+	// Warm up
+	time.Sleep(testWarmUpTime)
+
+	// Reset counters
+	noxu1Mutex.Lock()
+	noxu1ClientRequestLatencyCount = 0
+	noxu1ClientRequestLatencySum = 0
+	noxu1ClientRequestLatencySumSq = 0
+	noxu1Mutex.Unlock()
+	noxu2Mutex.Lock()
+	noxu2ClientRequestLatencyCount = 0
+	noxu2ClientRequestLatencySum = 0
+	noxu2ClientRequestLatencySumSq = 0
+	noxu2Mutex.Unlock()
+	earlierRequestExecutionSecondsSum, earlierRequestExecutionSecondsCount, earlierPLSeatUtilSum, earlierPLSeatUtilCount, err := getRequestExecutionMetrics(loopbackClient)
+	if err != nil {
+		t.Error(err)
+	}
+	time.Sleep(testTime) // after warming up, the test enters a steady state
+	laterRequestExecutionSecondsSum, laterRequestExecutionSecondsCount, laterPLSeatUtilSum, laterPLSeatUtilCount, err := getRequestExecutionMetrics(loopbackClient)
+	if err != nil {
+		t.Error(err)
+	}
+	if (earlierPLSeatUtilCount[priorityLevelNoxu1.Name] >= laterPLSeatUtilCount[priorityLevelNoxu1.Name]) || (earlierPLSeatUtilCount[priorityLevelNoxu2.Name] >= laterPLSeatUtilCount[priorityLevelNoxu2.Name]) {
+		t.Errorf("PLSeatUtilCount check failed: noxu1 earlier count %v, later count %v; noxu2 earlier count %v, later count %v",
+			earlierPLSeatUtilCount[priorityLevelNoxu1.Name], laterPLSeatUtilCount[priorityLevelNoxu1.Name], earlierPLSeatUtilCount[priorityLevelNoxu2.Name], laterPLSeatUtilCount[priorityLevelNoxu2.Name])
+	}
+	close(stopCh)
+
+	noxu1RequestExecutionSecondsAvg := (laterRequestExecutionSecondsSum[priorityLevelNoxu1.Name] - earlierRequestExecutionSecondsSum[priorityLevelNoxu1.Name]) / float64(laterRequestExecutionSecondsCount[priorityLevelNoxu1.Name]-earlierRequestExecutionSecondsCount[priorityLevelNoxu1.Name])
```

Comment:
I would make an additional type and function, so that all of these lines can look like `noxu1Avg := intervalAverage(t0, t1, plNoxu1.Name)` and the resulting type has members like `noxu1Avg.requestExecution` and `noxu1Avg.seats`.

(intervalAverage because what you are doing here is not just averaging)

The style as written makes it very easy to make a copy-paste error and very hard to visually detect one.